lib/r10k/source/git.rb

```diff
@@ -113,7 +119,17 @@ def desired_contents
 
   def branch_names
     opts = {:prefix => @prefix, :invalid => @invalid_branches, :source => @name}
-    @cache.branches.map do |branch|
+    branches = @cache.branches
+    if !@branch_filter.nil? && !@branch_filter.empty?
```

Comment:
The double negation here is funky. I would recommend a partially positive test instead such as...

`if @branch_filter && !@branch_filter.empty?`
or
`if @branch_filter && @branch_filter.size > 0`

Reply:
Opting for the former suggestion.